iceoryx_hoofs/test/moduletests/test_unix_domain_sockets.cpp

```diff
@@ -69,45 +75,282 @@ class UnixDomainSocket_test : public Test
     {
     }
 
-    static const size_t MaxMsgSize;
+    void createTestSocket(const UnixDomainSocket::UdsName_t& name)
+    {
+        static constexpr int32_t ERROR_CODE = -1;
+        struct sockaddr_un sockAddr;
+        int32_t sockfd;
+
+        memset(&sockAddr, 0, sizeof(sockAddr));
+        sockAddr.sun_family = AF_LOCAL;
+        strncpy(sockAddr.sun_path, name.c_str(), name.size());
+
+        iox::posix::posixCall(iox_socket)(AF_LOCAL, SOCK_DGRAM, 0)
+            .failureReturnValue(ERROR_CODE)
+            .evaluate()
+            .and_then([&](auto& r) { sockfd = r.value; })
+            .or_else([](auto&) {
+                std::cerr << "unable to create socket\n";
+                std::terminate();
+            });
+
+        iox::posix::posixCall(iox_bind)(sockfd, reinterpret_cast<struct sockaddr*>(&sockAddr), sizeof(sockAddr))
+            .failureReturnValue(ERROR_CODE)
+            .evaluate()
+            .or_else([](auto&) {
+                std::cerr << "unable to bind socket\n";
+                std::terminate();
+            });
+    }
+
+    void signalThreadReady()
+    {
+        doWaitForThread.store(false, std::memory_order_relaxed);
+    }
+
+    void waitForThread()
+    {
+        while (doWaitForThread.load(std::memory_order_relaxed))
+        {
+            std::this_thread::yield();
+        }
+    }
+
+    const std::chrono::milliseconds WAIT_IN_MS{10};
+    std::atomic_bool doWaitForThread{true};
     static constexpr uint64_t MaxMsgNumber = 10U;
     UnixDomainSocket server;
     UnixDomainSocket client;
 };
 
-const size_t UnixDomainSocket_test::MaxMsgSize = UnixDomainSocket::MAX_MESSAGE_SIZE;
 constexpr uint64_t UnixDomainSocket_test::MaxMsgNumber;
 
-TEST_F(UnixDomainSocket_test, NonBlockingModeNotSupported)
+TEST_F(UnixDomainSocket_test, UnlinkEmptySocketNameLeadsToInvalidChannelNameError)
 {
-    auto result = UnixDomainSocket::create(
-        goodName, IpcChannelMode::NON_BLOCKING, IpcChannelSide::SERVER, MaxMsgSize, MaxMsgNumber);
-    EXPECT_TRUE(result.has_error());
-    ASSERT_THAT(result.get_error(), Eq(IpcChannelError::INVALID_ARGUMENTS));
+    auto ret = UnixDomainSocket::unlinkIfExists(UnixDomainSocket::NoPathPrefix, "");
+    ASSERT_TRUE(ret.has_error());
+    EXPECT_THAT(ret.get_error(), Eq(IpcChannelError::INVALID_CHANNEL_NAME));
+}
+
+TEST_F(UnixDomainSocket_test, UnlinkEmptySocketNameWithPathPrefixLeadsToInvalidChannelNameError)
+{
+    auto ret = UnixDomainSocket::unlinkIfExists("");
+    ASSERT_TRUE(ret.has_error());
+    EXPECT_THAT(ret.get_error(), Eq(IpcChannelError::INVALID_CHANNEL_NAME));
+}
+
+TEST_F(UnixDomainSocket_test, UnlinkTooLongSocketNameWithPathPrefixLeadsToInvalidChannelNameError)
+{
+    UnixDomainSocket::UdsName_t longSocketName;
+    for (uint64_t i = 0U; i < UnixDomainSocket::LONGEST_VALID_NAME - strlen(UnixDomainSocket::PATH_PREFIX) + 1; ++i)
+    {
+        longSocketName.append(cxx::TruncateToCapacity, "o");
+    }
```

Comment:
maybe we need a `string::insert` method

Reply:
@FerdinandSpitzschnueffler yes we do. Also it would be nice if we could `concatenate`, `append` simple char's.

Reply:
Added these methods to https://github.com/eclipse-iceoryx/iceoryx/issues/208
Feel free to add more.